src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/InMemoryDirectoryInfo.cs

```diff
@@ -119,13 +119,11 @@ public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos()
 
         private bool IsRootDirectory(string rootDir, string filePath)
         {
-            if (!filePath.StartsWith(rootDir, StringComparison.Ordinal) ||
-                filePath.IndexOf(Path.DirectorySeparatorChar, rootDir.Length) != rootDir.Length)
-            {
-                return false;
-            }
+            int rootDirLength = rootDir.Length;
 
-            return true;
+            return filePath.StartsWith(rootDir, StringComparison.Ordinal) &&
+                (rootDir[rootDirLength - 1] == Path.DirectorySeparatorChar ||
+                filePath.IndexOf(Path.DirectorySeparatorChar, rootDirLength) == rootDirLength);
```

Comment:
Existing condition on line 126 was meant to avoid false positives when the path started with `rootDir` but it wasn't the complete name of the directory e.g:

```
var matcher = new Matcher();
matcher.AddInclude(@"**\*");

Console.WriteLine(matcher.Match(rootDir: @"C:\foo", file: @"C:\foo2\bar.cs").HasMatches); 
// prints false, which is correct.
```

However that same condition was interfering when `rootDir = @"C:\foo\"` since `filePath.IndexOf(Path.DirectorySeparatorChar, rootDirLength)` was always going to return the index of the separator of the next directory.

Therefore I added condition on line 125 in order to skip the length check when rootDir ends with a separator.

Reply:
It seems line 126 checks one char in position rootDirLength - why do we use whole IndexOf() method instead of simple index - filePath[rootDirLength] == Path.DirectorySeparatorChar - like in line 125?